block/slvalidator.go

```diff
@@ -90,6 +90,7 @@ func (v *SettlementValidator) ValidateStateUpdate(batch *settlement.ResultRetrie
 	daBlocks := []*types.Block{}
 	for _, batch := range daBatch.Batches {
 		daBlocks = append(daBlocks, batch.Blocks...)
+		types.LastReceivedDAHeightGauge.Set(float64(batch.EndHeight()))
```

Comment:
can be called **once** after `ValidateDaBlocks`?

Reply:
i think its fine in here, because this gauge exists to know whether its syncing from DA. i moved it to the validator flow, because it may sync from DA or not, but it will always fetch from DA when validating, so this gauge will be always updated with the latest batch fetched from DA. whether the last batch passes validation or not, imo is irrelevant for this.